internal/notifier/azure_eventhub.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2021 The Flux authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package notifier
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-amqp-common-go/v3/auth"
+	eventhub "github.com/Azure/azure-event-hubs-go/v3"
+	"github.com/fluxcd/pkg/runtime/events"
+)
+
+// AzureEventHub holds the eventhub client
+type AzureEventHub struct {
+	Hub *eventhub.Hub
+}
+
+// NewAzureEventHub creates a eventhub client
+func NewAzureEventHub(endpointURL, token, eventhubNamespace string) (*AzureEventHub, error) {
+	var hub *eventhub.Hub
+	var err error
+
+	// token should only be defined if JWT is used
+	if token != "" {
+		hub, err = newJWTHub(endpointURL, token, eventhubNamespace)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create a eventhub using JWT %v", err)
+		}
+	} else {
+		hub, err = newSASHub(endpointURL)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create a eventhub using SAS %v", err)
+		}
+	}
+
+	return &AzureEventHub{
+		Hub: hub,
+	}, nil
+}
+
+// Post all notification-controller messages to EventHub
+func (e *AzureEventHub) Post(event events.Event) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```

Comment:
You should add a check if the event is an update event as is done here.
https://github.com/fluxcd/notification-controller/blob/15d98c9afca937d01c64c7753df4615045f2fc65/internal/notifier/google_chat.go#L90